tests/python/pants_test/backend/jvm/tasks/jvm_compile/java/test_zinc_compile_integration.py

```diff
@@ -132,29 +134,72 @@ def test_no_zinc_file_manager(self):
         pants_run = self.run_test_compile(workdir, cachedir, target_spec, clean_all=True)
         self.assertEqual(0, pants_run.returncode)
 
-  def test_failed_hermetic_incremental_compile(self):
-    with temporary_dir() as cache_dir:
+  def test_hermetic_incremental_compile(self):
+    """
+    1) create a target containing two scala files
+    2) compile the target, which would be a full compile
+    3) modify a scala file slightly
+    4) recompile, and make sure the compile is incremental by checking the zinc outputs
+    """
+
+    def _create_file(path, value):
+      with safe_open(path, 'w') as f:
+        f.write(value)
+
+    def _create_a_target_containing_two_sources(_src_dir):
+      _srcfile = os.path.join(_src_dir, 'org', 'pantsbuild', 'cachetest', 'A.scala')
+      srcfile_b = os.path.join(_src_dir, 'org', 'pantsbuild', 'cachetest', 'B.scala')
+      buildfile = os.path.join(_src_dir, 'org', 'pantsbuild', 'cachetest', 'BUILD')
+      _lib_spec = os.path.join(os.path.basename(_src_dir), 'org', 'pantsbuild', 'cachetest')
+      _create_file(buildfile, 'scala_library()')
+      _srcfile_content = dedent("""
```

Comment:
Why so underscory with all the names?

Reply:
This is to distinguish them from variables outside the function, because in this case, the inner method can access the variables outside the method.

Reply:
Adjusted to have them all having underscore for consistency.